src/snowflake/snowpark/_internal/code_generation.py

```diff
@@ -0,0 +1,594 @@
+    ref_objects: Dict[str, Any],
+    classes_to_generate: List[type],
+) -> None:
+    """
+    Find objects referenced by functions including classes, methods, modules, global variables.
+
+    Args:
+        func: The target function to generate source code for.
+        to_import: set of name and alias pairs of direct imports which should be generated as "import xxx"
+            or "import xxx as yyy".
+        ref_objects: dict of objects referenced by the target function, key is the name and value is the object.
+        classes_to_generate: list of classes which are defined in the same module as the target function.
+            Code generation is required for these classes.
+    """
+    func_module_name = func.__module__
+    if isinstance(func, FunctionType):
+        get_func_references(func, ref_objects)
+        to_import.update(
+            extract_submodule_imports(
+                func, [v for v in ref_objects.values() if isinstance(v, ModuleType)]
+            )
+        )
+        for v in [
+            v
+            for v in ref_objects.values()
+            if inspect.isclass(v) and v.__module__ == func_module_name
+        ]:
+            get_class_references(v, func, ref_objects, classes_to_generate)
+    elif isinstance(func, BuiltinFunctionType):
+        if func_module_name != builtins.__name__:
+            to_import.add((func_module_name, func_module_name))
+
+
+def resolve_target_func_referenced_objects_by_type(
+    func: Union[FunctionType, BuiltinFunctionType],
+    to_import: Set[Tuple[str, str]],
+    to_import_from_module: Dict[str, Set[Tuple[str, str]]],
+    ref_objects: Dict[str, Any],
+    code_as_comment: bool,
+) -> Tuple[str, str]:
+    """
+    Deal with the referenced objects by types, handles modules/classes/methods/global variables, generate source code
+    for referenced functions defined in the same module as the target function's and referenced variables.
+
+    Args:
+        func: The target function to generate source code for.
+        to_import: set of name and alias pairs of direct imports which should be generated as "import xxx"
+            or "import xxx as yyy".
+        to_import_from_module: dict of import information, key is the module name with value being the set of
+            name and alias paris of imported objects which should be generated as "from xxx import yyy" or
+            "from xxx import yyy as zzz".
+        ref_objects: dict of objects referenced by the target function, key is the name and value is the object.
+        code_as_comment: Whether the code will be generated as comment.
+
+    Returns:
+        A tuple of two strings, the first one is the source code of referenced functions defined in the same module
+        as the target function's, and the second is the source code of referenced variables.
+    """
+    func_module_name = func.__module__
+    func_text = ""
+    global_vars_text = ""
+    for name, obj in ref_objects.items():
```

Comment:
It seems like we are doing different actions based on different predicates and there are very few if any shared code, how about do early return style for all:

```
for ...:
  if ...:
    ...
    continue
  if ...:
    ...
    continue
...
```

Reply:
I personally prefer `if-elif-else` style.

my personal experience of early return is that the executions results are chained sequentially in different clauses while here we are dealing with completely different predicates.